statgen/pheget
# vuejs, bookmarkable url views, and ui tweaks

## Ticket
Replaces #48 

## Purpose
The app is far more interactive than it was designed for. In cases where an option can be set in multiple places (like a radio button, url, or dropdown), we are writing a lot of redundant event code and it has become hard to keep all parts of the UI in sync.

This PR preserves the flask backend (data APIs for LZ, and view APIs with the data for a given URL), but it rewrites the template logic in vue.js/ vue-router. This enables better dynamic behavior and keeps all parts of the page consistent.

The JS build tool takes some time to set up once but after that is fairly convenient. It will automatically handle transpiling (so we can use modern JS syntax), and tracks asset versions so we are no longer dependent on inconsistent CDNs.

It also adds an about page etc as a home for some of the help text.

## How to test this
See demo page (a staging server) to start:
https://eqtl.occsci.com/region?chrom=19&start=448506&end=528506&gene_id=ENSG00000129946.10&tissue=Brain_Cortex&y_field=log_pvalue

## Deployment / configuration notes
We are moving to a SPA approach, which changes the way we think about deployment and development.

This has large impacts on server configuration; a working apache sample and new development instructions are provided.
